game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java

```diff
@@ -1111,6 +1095,16 @@
           }
         }
 
+        // Mimic the legacy behavior and force the parties to be non-allied modifying either
+        // attacker or defender like the old code did). This legacy behavior was requested in #9532.
```

Comment:
A potential maintainer reading this comment might be curious what "legacy behavior" refers to, it's not clear without context of a PR. Requiring a reader to reference a PR is a smell, PRs are ephemeral (and subsequent PRs can change things).

This comment perhaps should focus on explaining what we are doing in this code, what the behavior is, rather than it is simply "legacy" (albeit, still the current behavior).